Update Site Logo block description to be concise

annezazu opened the pull request:
The current site logo description is quite unnecessarily verbose. This minor update simplifies the language used to make it more concise while still attempting to get the right messaging across. This came up as part of the [ninth call for testing for the FSE Outreach Program](https://make.wordpress.org/test/2021/08/11/fse-program-testing-call-9-handling-highered-headers/#comment-1928). Hat tip to @richtabor for pointing it out and hat tip to @kellychoffman for co-labing on the language. 

## Description

Update to the site logo block sidebar description

## How has this been tested?


## Screenshots

Before:

<img width="285" alt="Screen Shot 2021-09-01 at 2 42 47 PM" src="https://user-images.githubusercontent.com/26996883/131726439-930cd0c2-91b9-4a6c-beae-dbf4972bccaf.png">

After:

<img width="284" alt="Screen Shot 2021-09-01 at 3 20 37 PM" src="https://user-images.githubusercontent.com/26996883/131731124-7c849027-a446-484a-81cd-bdb4dacd541c.png">


## Types of changes

Enhancement

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

annezazu committed:
Update Site Logo block description to be concise

The current site logo description is quite unnecessarily verbose. This minor update simplifies the language used to make it more concise while still attempting to get the right messaging across.

annezazu committed:
Made language even more concise

mtias commented:
I understand the intention to trim copy but I'm not a fan of leaving things vague in the name of brevity when we could be using the opportunity to explain what things are. A logo is not something you'd be using day to day, so it can afford more explanatory descriptions and guidance. Not saying the current wording is perfect, and I'd agree it's a bit lengthy in excess. Until we have a flow for also updating a site icon from the site logo, for example, I think some clarification is useful.

annezazu commented:
This update feels about more than just brevity but the description and PR title doesn't quite capture that. Let's add in a bit more context though as I agree that without a site icon flow is in place it's an important distinction. How does something like this sound? 

> Display a graphic to represent this site. Update the block, and the changes apply everywhere it’s used. This is different than the site icon, which is the smaller image visible in your dashboard, browser tabs, etc used to help others recognize this site.

annezazu committed:
Add in educational context to the site logo description

Since there isn't an option to add a site icon in FSE, it's important to add extra context about what this is vs what it isn't.
